src/modules/ethereum/loanTransactions.js

```diff
@@ -240,88 +268,110 @@ export async function collectLoansTx(loansToCollect) {
-        .utc()
-        .unix();
-    let loanState = null;
-    let isDue = false;
-    let isRepayable = false;
-    let isCollectable = false;
-    switch (solidityLoanState) {
-        case 0:
-            if (maturity < currentTime) {
-                loanState = 21;
-                isCollectable = true;
-                loanStateText = "Defaulted (not yet collected)";
-            } else if (maturity - currentTime < 24 * 60 * 60 * 2) {
-                /* consider it due 2 days before */
-                isDue = true;
-                isRepayable = true;
-                loanStateText = "Payment Due";
-                loanState = 5;
-            } else {
-                isRepayable = true;
-                loanState = 0;
-                loanStateText = "Open";
-            }
-            break;
-        case 1:
-            loanState = 10;
-            loanStateText = "Repaid";
-            break;
-        case 2:
-            loanState = 20;
-            loanStateText = "Defaulted";
-            break;
-        default:
-            loanStateText = "Invalid state";
+    // TODO: resolve timing of loanManager refresh in order to get chunkSize & loanCount from loanManager:
+    const [chunkSize, loanCount] = await Promise.all([
+        loanManager.CHUNK_SIZE().then(res => res.toNumber()),
+        loanManager.getLoanCountForAddress(account).then(res => res.toNumber())
+    ]);
+    // const chunkSize = store.getState().loanManager.info.chunkSize;
+    // const loanCount = await loanManager.getLoanCountForAddress(account);
+
+    let loans = [];
+
+    const queryCount = Math.ceil(loanCount / chunkSize);
+
+    for (let i = 0; i < queryCount; i++) {
+        const loansArray = await loanManager.getLoansForAddress(account, i * chunkSize);
+        loans = loans.concat(parseLoans(loansArray));
     }
-    // TODO: refresh this reguraly? maybe move this to a state and add a timer?
 
-    const loan = {
-        loanId,
-        borrower,
-        loanState,
-        solidityLoanState,
-        loanStateText,
-        collateralEth: bn_collateralAmount / ONE_ETH,
-        repaymentAmount: bn_repaymentAmount / decimalsDiv,
-        loanAmount: bn_loanAmount / decimalsDiv,
-        interestAmount: bn_interestAmount / decimalsDiv,
-        term,
-        termText: moment.duration(term, "seconds").humanize(),
-        disbursementDate,
-        disbursementDateText: moment.unix(disbursementDate).format("D MMM YYYY HH:mm:ss"),
-        maturity,
-        maturityText: moment.unix(maturity).format("D MMM YYYY HH:mm"),
-        isDue,
-        isRepayable,
-        isCollectable
-    };
-    return loan;
+    return loans;
+}
+
+function parseLoans(loansArray) {
+    const decimalsDiv = store.getState().augmintToken.info.decimalsDiv;
+
+    const loans = loansArray.reduce((parsed, loan) => {
+        // [loanId, collateralAmount, repaymentAmount, borrower, productId, state, maturity, disbursementTime, loanAmount, interestAmount ]
```

Comment:
temporary comment leftover, please remove!

Reply:
removed, pushing it soon